packages/bpk-dialog-utils/BpkDialogWrapper.tsx

```diff
@@ -0,0 +1,153 @@
+const dialogSupported = typeof HTMLDialogElement === 'function';
+
+const setPageProperties = ({ isDialogOpen }: DialogProps) => {
+  document.body.style.overflowY = isDialogOpen ? 'hidden' : 'visible';
+
+  if (!dialogSupported) {
+    document.body.style.position = isDialogOpen ? 'fixed' : 'relative';
+    document.body.style.width = isDialogOpen ? '100%' : 'auto';
+  }
+};
+
+export const BpkDialogWrapper = (props: Props) => {
+  const {
+    ariaLabelledby,
+    children,
+    closeOnEscPressed = false,
+    closeOnScrimClick = false,
+    dialogClassName = '',
+    exiting = false,
+    id,
+    isOpen,
+    onClose,
+    timeout = {appear: 0, exit: 0},
+    transitionClassNames = {}
+  } = props;
+
+  const ref = useRef<HTMLDialogElement>(null);
+
+  useEffect(() => {
+    const dialog = document.getElementById(`${id}`);
+    const dialogWithPolyfill = document.getElementById(`${id}-polyfill`);
+
+    const handleBackdropClick = (modal: HTMLElement | null) => {
+      if (closeOnScrimClick && modal) {
+        modal.addEventListener('click', (event: MouseEvent | PointerEvent) => {
+          const { target } = event;
+
+          if (target === modal) {
+            onClose(event, {source: "DOCUMENT_CLICK"});
+            event.stopPropagation();
+          }
+        });
+      }
+    };
+
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (closeOnEscPressed && event.key === 'Escape') {
+        onClose(event, {source: "ESCAPE"});
+      }
+      event.stopPropagation();
+    };
+
+    if (isOpen) {
+      // There is a bug on older versions of browser using chromium (chrome, firefox, edge >114) where the dialog got an open attribute even before it is opened.
+      // Therefore, when trying to open it, it crashes and log an error mentioning the dialog has already an open attribute.
+      ref.current?.removeAttribute('open');
+      ref.current?.showModal?.();
+
+      if (dialogWithPolyfill) {
+        handleBackdropClick(dialogWithPolyfill);
+        window.addEventListener('keydown', handleKeyDown);
+      }
+      handleBackdropClick(dialog);
+    } else {
+      ref.current?.close?.();
+    }
+
+    setPageProperties({ isDialogOpen: isOpen });
+    return () => {
+      setPageProperties({ isDialogOpen: false })
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [id, isOpen, onClose, closeOnEscPressed, closeOnScrimClick]);
+
+  return isOpen ? (
+    <div
+      className={getClassName(
+        'bpk-dialog-wrapper',
+        dialogSupported ? '' : 'bpk-dialog-wrapper--polyfill',
+      )}
+    >
+      {!dialogSupported && (
+        <div
+          id={`${id}-polyfill`}
+          className={getClassName('bpk-dialog-wrapper--backdrop')}
+          data-open={isOpen}
+        />
+      )}
+      <CSSTransition
+        classNames={transitionClassNames}
+        in={!exiting}
+        appear={!exiting}
+        exit={exiting}
+        timeout={timeout}
+      >
+        <dialog
```

Comment:
Sharing this comment here as I think this will be important: https://github.com/Skyscanner/backpack/pull/3209#pullrequestreview-1890544627

Would be worth us running a test in Safari 14 to see how this works and make sure it is safe and doesn't break for folks

Reply:
If the code was copied form bpkmodalv2, there is a polyfill for old browser that don't support the dialog element.

Reply:
Yes, BpkDialogWrapper is essentially a copy of the BpkModalV2 dialog, with optional animations and disabling ESC or background click close.